server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1302,7 +1320,9 @@ public Node start() throws NodeValidationException {
             injector.getInstance(MetaStateService.class),
             injector.getInstance(MetadataIndexUpgradeService.class),
             injector.getInstance(MetadataUpgrader.class),
-            injector.getInstance(PersistedClusterStateService.class)
+            injector.getInstance(PersistedClusterStateService.class),
+            injector.getInstance(RemoteClusterStateService.class),
```

Comment:
would throw error, in case the instance is null. did you run it with remote state disabled as well?